featurebyte/api/feature_list.py

```diff
@@ -364,11 +365,16 @@ def _initialize_feature_objects_and_items(cls, values: dict[str, Any]) -> dict[s
 
                     # deserialize feature record into feature object
                     feature = Feature.from_persistent_object_dict(object_dict=feature_dict)
-                    items.append(feature)
-                    feature_objects[feature.name] = feature
+                    feature_id_to_object[str(feature.id)] = feature
                     progress_bar.text = feature.name
                     progress_bar()  # pylint: disable=not-callable
 
+                # preserve the order of features
```

Comment:
nit: Maybe we can move the initialisation of `items` and `feature_objects` down here if they are not referenced earlier.

Reply:
Will fix it in a separate PR.